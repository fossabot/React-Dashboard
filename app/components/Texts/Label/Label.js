import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import classes from './Label.scss';

const Label = ({ className, children, htmlFor }) => (
  <label className={classnames(className, classes.element)} htmlFor={htmlFor}>{children}</label>
);

Label.propTypes = {
  className: PropTypes.string,
  htmlFor: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.element,
  ]).isRequired,
};

Label.defaultProps = {
  className: null,
  htmlFor: '',
};

export default Label;
