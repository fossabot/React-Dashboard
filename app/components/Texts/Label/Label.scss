.element {
  font-family: "Source Sans Pro";
  font-size: 14px;
  font-weight: 500;
}
