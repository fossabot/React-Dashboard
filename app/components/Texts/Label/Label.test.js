import { shallow } from 'enzyme';
import Label from './Label';

describe('Label', () => {
  it('should render children', () => {
    const component = shallow(<Label debug>email</Label>);
    expect(component.contains('email')).toEqual(true);
  });
});
