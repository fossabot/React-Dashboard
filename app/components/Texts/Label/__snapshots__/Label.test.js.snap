// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label should render correctly in "debug" mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Label
    debug={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "element",
    },
    "ref": null,
    "rendered": null,
    "type": "label",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "element",
      },
      "ref": null,
      "rendered": null,
      "type": "label",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
